src/script/script.cpp

```diff
@@ -258,6 +324,44 @@ bool CScript::IsPushOnly() const
     return this->IsPushOnly(begin());
 }
 
+
+bool CScript::IsUnspendable() const
+{
+    if (size() > MAX_SCRIPT_SIZE) {
+        return true;
+    } else if (this->IsPayToPublicKey()) {
+        std::vector<unsigned char> pubkey(this->begin() + 1, this->end() - 1);
+        if(!CPubKey(pubkey).IsFullyValid()){
+            return true;
+        }
+    } else if (this->IsPayToMultisig()) {
+        CScript::const_iterator pc = this->begin() + 1;
```

Comment:
Seems like it would be better to have `this->IsInvalidMultisig()` and move this code all into one place? Something like:

```c++
if (size() < 3) return false;
if (back() != OP_CHECKMULTISIG && back() != OP_CHECKMULTISIGVERIFY) return false;
int n;
if (!IsOpN(*(end() - 2), n ) return false;
int m;
if (!IsOpN(front(), m) return false;
std::vector<Span<unsigned char>> keys;
keys.resize(n);
// parse into keys, finish checking it's well-formed
if (m > n) return true;
int good = 0;
int max_good = n;
for (k : keys) {
    if (CPubKey(k).IsFullyValid()) {
        ++good;
    } else {
        --max_good;
    }
    if (good >= m) return false;
    if (max_good < m) return true;
}
return false; // unreachable